host/src/proof.rs

```diff
@@ -79,6 +79,12 @@ impl ProofActor {
     }
 
     pub async fn cancel_task(&mut self, key: ProofTaskDescriptor) -> HostResult<()> {
+        // Remove cancelling task from pending tasks
```

Comment:
no need here because cancel eventually make a signal to task process and break the loop there, so the pending will be cleared by complete signal there (correct me if I forget sth), anyway, this cancel is kind of tricky and better to refactory later on.

Reply:
Agree that `cancel` needs some more re-design and test coverage.